packages/lesswrong/components/seasonal/annualReview/AnnualReviewSidebarBanner.tsx

```diff
@@ -0,0 +1,291 @@
+    borderRadius: '3px',
+    textAlign: 'center',
+    padding: '8px 14px',
+    cursor: 'pointer',
+    display: 'inline-block',
+    textShadow: 'none',
+    '&:hover': {
+      textDecoration: 'none',
+      opacity: 1,
+    },
+    fontWeight: '500',
+    fontSize: '22px',
+  }
+}));
+
+function formatPhaseDateRange(startDate: moment.Moment, endDate: moment.Moment) {
+  if (startDate.month() === endDate.month()) {
+    return `${startDate.format('MMMM D')}-${endDate.format('D')}`;
+  }
+  return `${startDate.format('MMM D')} - ${endDate.format('MMM D')}`;
+}
+
+const commonIntro = "Each December, the LessWrong community reviews the best blogposts of yesteryear.";
+
+interface PhaseConfig {
+  phaseName: string;
+  phaseDescription: string;
+  buttonLabel: string;
+  getDateRange: () => { start: moment.Moment; end: moment.Moment };
+  getButtonLink: () => string;
+}
+
+const nominatePostsLink = `/nominatePosts/${REVIEW_YEAR}?${qs.stringify(allPostsParams(REVIEW_YEAR))}`;
+const reviewPostsLink = `/reviewVoting`;
+
+const phaseConfigs: Partial<Record<ReviewPhase, PhaseConfig>> = {
+  NOMINATIONS: {
+    phaseName: "Nomination",
+    phaseDescription: `In the nomination phase, we identify posts worthy of consideration in the review.`,
+    buttonLabel: "Nominate Now",
+    getDateRange: () => ({
+      start: getReviewStart(REVIEW_YEAR),
+      end: getNominationPhaseEndDisplay(REVIEW_YEAR),
+    }),
+    getButtonLink: () => nominatePostsLink,
+  },
+  REVIEWS: {
+    phaseName: "Discussion",
+    phaseDescription: `In the discussion phase, we review and debate posts. Posts that receive one review move to the final Voting Phase.`,
+    buttonLabel: "Review Now",
+    getDateRange: () => ({
+      start: getNominationPhaseEnd(REVIEW_YEAR),
+      end: getReviewPhaseEndDisplay(REVIEW_YEAR),
+    }),
+    getButtonLink: () => reviewPostsLink,
+  },
+  VOTING: {
+    phaseName: "Voting",
+    phaseDescription: `In the Final Voting phase, we do a full voting pass. The outcome determines Best of LessWrong results.`,
+    buttonLabel: "Vote Now",
+    getDateRange: () => ({
+      start: getReviewPhaseEnd(REVIEW_YEAR),
+      end: getVotingPhaseEndDisplay(REVIEW_YEAR),
+    }),
+    getButtonLink: () => reviewPostsLink,
+  },
+  RESULTS: {
+    phaseName: "Results",
+    phaseDescription: `Voting is complete! Check out the results and see which posts made it into Best of LessWrong.`,
+    buttonLabel: "View Results",
+    getDateRange: () => ({
+      start: getVotingPhaseEndDisplay(REVIEW_YEAR),
+      end: getResultsPhaseEnd(REVIEW_YEAR),
+    }),
+    getButtonLink: () => reviewPostsLink,
+  },
+};
+
+export const AnnualReviewSidebarBanner = () => {
+  const classes = useStyles(styles);
+  
+  const activePhase = getReviewPhase(REVIEW_YEAR);
+  
+  if (activePhase === "UNSTARTED" || activePhase === "COMPLETE") {
+    return null;
+  }
+
+  const config = phaseConfigs[activePhase];
+  if (!config) return null;
```

Comment:
Have you checked that this entire component looks ok during all 4 phases?  (In terms of text alignment/etc)

Reply:
yes

Reply:
but need to check darkmode